File: DotNET/restTemplateSimple.cs
// This sample requires .NET 7.1 or later, and Newtonsoft.Json.

using System;
using System.Net.Http;
using System.Text;
using Newtonsoft.Json;

namespace SampleFunction
{
    class Program
    {
        static void SampleFunction()
        {
            string host = "https://api.cognitive.microsofttranslator.com";
            string route = "/translate?api-version=3.0&to=de&to=it";
            // Add your subcription key.
            string subscriptionKey = "YOUR_SUBSCRIPTION_KEY";
            /*
             * Creates the body of the request. This sample is structure for
             * Translator Text, which is an array of Text objects.
             */
            System.Object[] body = new System.Object[] { new { Text = @"Hello world." } };
            var requestBody = JsonConvert.SerializeObject(body);

            using (var client = new HttpClient())
            using (var request = new HttpRequestMessage())
            {
                // Set the method to POST
                request.Method = HttpMethod.Post;
                // Construct the full URI
                request.RequestUri = new Uri(host + route);
                // Add the serialized JSON object to your request
                request.Content = new StringContent(requestBody, Encoding.UTF8, "application/json");
                // Add the authorization header
                request.Headers.Add("Ocp-Apim-Subscription-Key", subscriptionKey);
                // Send request, get response
                var response = client.SendAsync(request).Result;
                var jsonResponse = response.Content.ReadAsStringAsync().Result;
                // Print the response
                Console.WriteLine(jsonResponse);
                Console.WriteLine("Press any key to continue.");
            }
        }
        /*
         * The main function makes a post request and prettifies the response.
        */
        static void Main(string[] args)
        {
            SampleFunction();
            Console.ReadLine();

        }
    }
}

File: Java/restTemplateSimple.java
/*
 * This sample requires JDK 7 or later, OkHttp, and Gson. Gradle files are
 * available in the Gradle-Files directory.
 */
import java.io.*;
import java.net.*;
import java.util.*;
import com.google.gson.*;
import com.squareup.okhttp.*;

public class Translate {
    // Add your subscription key.
    String subscriptionKey = "YOUR_SUBSCRIPTION_KEY";
    String url = "https://api.cognitive.microsofttranslator.com/translate?api-version=3.0&to=de,it";

    // Instantiates the OkHttpClient.
    OkHttpClient client = new OkHttpClient();

    // This function performs a POST request.
    public String post() throws IOException {
        MediaType mediaType = MediaType.parse("application/json");
        /*
         * Creates the body of the request. This sample is structure for
         * Translator Text, which is an array of Text objects.
         */
        RequestBody body = RequestBody.create(mediaType,
                "[{\n\t\"Text\": \"Welcome to Microsoft Translator. Guess how many languages I speak!\"\n}]");
        Request request = new Request.Builder()
                .url(url).post(body)
                // Add additional headers here.
                .addHeader("Ocp-Apim-Subscription-Key", subscriptionKey)
                .addHeader("Content-type", "application/json").build();
        Response response = client.newCall(request).execute();
        return response.body().string();
    }

    /*
     * This function prettifies the json response. This is not required, but
     * makes the requests human readable for quickstarts. If you remove this
     * function, make sure to remove the preffity function in main.
     */
    public static String prettify(String json_text) {
        JsonParser parser = new JsonParser();
        JsonElement json = parser.parse(json_text);
        Gson gson = new GsonBuilder().setPrettyPrinting().create();
        return gson.toJson(json);
    }
    /*
     * The main function makes a post request and prettifies the response.
    */
    public static void main(String[] args) {
        try {
            Translate translateRequest = new Translate();
            String response = translateRequest.post();
            System.out.println(prettify(response));
        } catch (Exception e) {
            System.out.println(e);
        }
    }
}
